document custom dictionaries (#4771)
mattermost/docs: source/install/desktop-custom-dictionaries.rst

This feature is available in the Mattermost Desktop App from v4.7.1 on Windows and Linux. Custom dictionaries are not supported in the Mattermost Desktop App on macOS.

Setting up the dictionaries
---------------------------

The Mattermost Desktop App uses [Hunspell Dictionary definitions](https://hunspell.github.io/) for Windows and Linux. For macOS, dictionary definitions are provided by Apple within the operating system itself. A quick way to access default dictionary definitions is to obtain a copy of `hunspell_dictionaries.zip` from the [latest electron release](https://github.com/electron/electron/releases/latest).

Once downloaded, the dictionary definitions need to be extracted and converted to lowercase because chromium expects `en-us` rather than `en-US...`. These defitions are ready to use and would match the usage of using Chromium's CDN ones without accesing it, but they could also be modified to fit specific needs.

Serve the files from a web server, and write down the URL to the root folder of the dictionaries for using it in the Desktop App.

Configuring the Desktop App
---------------------------

> Open the Mattermost Desktop App, go to **… > File > Settings**. Under **App Options**, and ensure that the **Check Spelling** option is enabled. Below this setting there is a link named **Use and alternative dictionary URL**. Use this option to specify the URL to the root folder of the dictionaries, then select `Save`.

Reviewer: ```suggestion
Open the Mattermost Desktop App, go to **… > File > Settings**. Under **App Options**, ensure that the **Check Spelling** option is enabled. Below this setting there is a link named **Use an alternative dictionary URL**. Use this option to specify the URL to the root folder of the dictionaries, then select **Save**.
```

Desktop App Custom Dictionaries

> ===========================

Reviewer 2: ```suggestion
=====================================
```

Mattermost Desktop App users running Windows or Linux can download dictionary definitions from a custom URL. If custom dictionaries aren't specified, default dictionary definitions are obtained automatically from Chromium's CDNs.

> such a service, dictionaries definitions will be automatically obtained from Chromium's CDNs.

Reviewer 2: We can safely delete this line of text: "such a service, dictionaries definitions will be automatically obtained from Chromium's CDNs."

Author: oh! removed the next one but somehow I missed this one, will fix it now

This feature is available in the Mattermost Desktop App from v4.7.1 on Windows and Linux. Custom dictionaries are not supported in the Mattermost Desktop App on macOS.

Setting up the dictionaries
---------------------------

The Mattermost Desktop App uses [Hunspell Dictionary definitions](https://hunspell.github.io/) for Windows and Linux. For macOS, dictionary definitions 
are provided by Apple within the operating system itself. A quick way to access default dictionary definitions is to obtain a copy of `hunspell_dictionaries.zip` from the [latest 
electron release](https://github.com/electron/electron/releases/latest).

Once downloaded, the dictionary definitions need to be extracted and converted to lowercase because chromium expects `en-us` rather than `en-US...`.
These defitions are ready to use and would match the usage of using Chromium's CDN ones without accesing it, but they could also be modified to fit specific needs.

Serve the files from a web server, and write down the URL to the root folder of the dictionaries for using it in the Desktop App.

Configuring the Desktop App
---------------------------

> Access Mattermost Desktop App Preferences by opening the Desktop App, then going to **… > File > Settings**. Under **App Options**, ensure that the **Check Spelling** option is enabled. Below this setting there is a link 

Reviewer 2: ```suggestion
Open the Mattermost Desktop App, go to **… > File > Settings**. Under **App Options**, and ensure that the **Check Spelling** option is enabled. Below this setting there is a link 
```

Desktop Custom dictionaries
===========================

Mattermost Desktop allows users to download their dictionary definitions from a custom URL. If the user doesn't want to provide
such a service, dictionaries definitions will be automatically obtained from Chromium's CDNs.

> This feature enables users to skip using Chromium's CDNs as well as to provide customized dictionaries definitions.

Reviewer 2: ```suggestion
This feature allows users to provide customized dictionary definitions instead of using Chromium's CDNs.
```

This feature is available since v4.7.1 on Windows and Linux.

Setting up the dictionaries
---------------------------

> The desktop app uses [Hunspell Dinctionary definitions](https://hunspell.github.io/) for Windows and Linux (for macOS definitions 

Reviewer 2: ```suggestion
The Mattermost Desktop App uses [Hunspell Dictionary definitions](https://hunspell.github.io/) for Windows and Linux. For macOS, dictionary definitions 
```

> are provided by the OS itself). A quick way to get them is to obtain a copy of `hunspell_dictionaries.zip` from the [latest 

Reviewer 2: ```suggestion
are provided by Apple within the operating system itself. A quick way to access default dictionary definitions is to obtain a copy of `hunspell_dictionaries.zip` from the [latest 
```

electron release](https://github.com/electron/electron/releases/latest).

> Once obtained it needs to be extracted and converted to lowercase, as chromium won't try to access `en-US...` but `en-us...`.

Reviewer 2: ```suggestion
Once downloaded, the dictionary definitions need to be extracted and converted to lowercase because chromium expects `en-us` rather than `en-US...`.
```

Reviewer 2: Does it make sense to mention that once the default dictionary definitions are downloaded, users can then modify them as needed with their own definitions?

Author: i was just trying to provide a minimum working solution, but I guess that's doable as well :)

> Serve the files from a webserver and copy the URL to the root folder of the dictionaries.

Reviewer 2: ```suggestion
Serve the files from a web server, and copy the URL to the root folder of the dictionaries.
```

Reviewer 2: Can we expand on this step a bit further? Where does the URL need to be copied to, specifically?

Author: oh, it is to be used in the configuring section, should I use a different word for that? my first option was memorize or write down, but seemed less likely that copying for later use.

> Configuring the desktop app

Reviewer 2: ```suggestion
Configuring the Desktop App
```

---------------------------

> Open the desktop app and go to the settings tab and ensure that check spelling is turned on. Below this setting there is a link 

Reviewer 2: ```suggestion
Access Mattermost Desktop App Preferences by opening the Desktop App, then going to **… > File > Settings**. Under **App Options**, ensure that the **Check Spelling** option is enabled. Below this setting there is a link 
```

> named `Use and alternative dictionary URL`, clicking on it will reveal an input box. Fill it with the URL to the root folder of 

Reviewer 2: ```suggestion
named **Use and alternative dictionary URL**. Use this option to specify the URL to the root folder of 
```

> the dictionaries and click `Save`.

Reviewer 2: ```suggestion
the dictionaries, then select `Save`.
```

> Removing the customized dictionaries

Reviewer 2: ```suggestion
Removing customized dictionaries
```

------------------------------------

> Open the preferences of the desktop app and below the input box there is a link `Revert to default`, clicking on it will disable 

Reviewer 2: To stop using the specified dictionary definitions, select the **Revert to default** preference option under **App Options** to use Chromium's CDNs instead.
